src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipService.java

```diff
@@ -49,9 +51,18 @@ public String createLimitedPartnership(Transaction transaction,
         LimitedPartnershipSubmissionDao dao = mapper.dtoToDao(limitedPartnershipSubmissionDto);
         dao.setCreatedAt(LocalDateTime.now());
         dao.setUserId(userId);
+
+        
+        // Create the self-link
+        var selfLink = String.format("/transactions/%s/limited-partnership/partnership/%s", transaction.getId(), dao.getId());
+        var links = new HashMap<String, String>();
+        links.put("self", selfLink);
+        dao.setLinks(links);
```

Comment:
What is the purpose of this code block?  Looks redundant to me.

Reply:
The creation and setting of the self-link are necessary steps to ensure that the DAO is fully populated with all required information before it is saved to the database. Therefore, this part of the code is not redundant but essential for the correct functioning of the createLimitedPartnership method.

Reply:
But you are not saving this dao object to the database. You are populating it with the self link and then doing nothing with it.

Reply:
Just had another look. I can see that the dao is being inserted into the database - but as per previous comments, the self link needs the submissionId so the record needs to be updated with the self link **after** the records has been created. So my comment still stands this code appears to be redundant.